js/window-close/demos/open-close-window/index.html

```diff
@@ -0,0 +1,99 @@
+<!DOCTYPE html>
+<html lang="ru">
+<head>
+  <title>Пример window.close() — Дока</title>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="preconnect" href="https://fonts.googleapis.com">
+  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+  <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500&display=swap">
+  <style>
+    *,
+    *::before,
+    *::after {
+      margin: 0;
+      padding: 0;
+      box-sizing: border-box;
+    }
+
+    html,
+    body {
+      scrollbar-gutter: stable;
+    }
+
+    html {
+      height: 100vh;
+      color-scheme: dark;
+      font-size: 18px;
+    }
+
+    body {
+      min-height: 100vh;
+      padding: 50px;
+      display: flex;
+      flex-direction: column;
+      justify-content: center;
+      align-items: center;
+      gap: 4rem;
+      background-color: #18191C;
+      color: #FFFFFF;
+      font-family: "Roboto", sans-serif;
+    }
+
+    .button {
+      min-width: 210px;
+      padding: 9px 15px;
+      border: 2px solid transparent;
+      border-radius: 6px;
+      font-size: 1rem;
+      font-weight: 300;
+      font-family: inherit;
+      transition: background-color 0.2s linear;
+      color: #000000;
+      background-color: #C56FFF;
+    }
+
+    .button:hover {
+      background-color: #FFFFFF;
+      cursor: pointer;
+      transition: background-color 0.2s linear;
+    }
+
+    .button:disabled {
+      opacity: 0.5;
+      pointer-events: none;
+    }
+  </style>
+</head>
+<body>
+  <button id="buttonOpen" class="button">Открыть окно</button>
+  <button id="buttonClose" class="button" disabled>Закрыть окно</button>
+  <script>
+    const buttonOpen = document.querySelector('#buttonOpen')
+    const buttonClose = document.querySelector('#buttonClose')
+
+    const openWindow = () => {
+      const newWindow = window.open('https://doka.guide/', '_blank', 'top=40, left=40, width=500, height=400')
+
+      return newWindow
+    }
+    const closeWindow = (windowProxy) => windowProxy.close()
+
+    let currentWindow
+
+    buttonOpen.addEventListener('click', function (event) {
+      buttonOpen.disabled = true
+      buttonClose.disabled = false
+
+      currentWindow = openWindow()
+    })
```

Comment:
Есть такая штука, что если нативным крестиком закрываем окно, то кнопки не меняют своё состояние. Набросал код с интервалом, который будет реагировать на закрытие окна. Давай добавим, как смотришь на это?

```suggestion
    const closeWindow = (windowProxy) => windowProxy.close()
    let currentWindow
    const watchTheWindowClose = () => {
      const timer = setInterval(function() {
        if (currentWindow.closed) {
          clearInterval(timer)
        
          buttonClose.disabled = true
          buttonOpen.disabled = false
        }
      }, 100)
    }

    buttonOpen.addEventListener('click', function (event) {
      buttonOpen.disabled = true
      buttonClose.disabled = false

      currentWindow = openWindow()
      
      watchTheWindowClose()
    })
```

Reply:
Мне кажется, что это лишнее. Статья описывает закрытие окна, зачем же нам демонстрировать слежение за ним?

Reply:
Потому что без слежения будет баг, а демка с багом это не красиво. Тем более я привёл пример кода выше, так что дополнительно кодить ничего не нужно.

То есть ещё раз: да, мы демонстрируем закрытие окна, ты совершенно прав. Но в этой демонстрации есть небольшой баг, который я предлагаю легко исправить.

Reply:
> Потому что без слежения будет баг,

а в чём баг ?
Я не то что пытаюсь упрямится, но хочется понять.
Выполнять код слежения раз в 100 ms - ну это ж на любителя )

Reply:
Конечно, повторю без проблем) Если нативным крестиком закрываем окно, то кнопки не меняют своё состояние. Кнопка закрыть окно **остаётся доступной для нажатия**, а кнопка открыть окно заблокирована. Поэтому, чтобы окно снова открыть, нужно сначала нажать на закрыть (хотя фактически мы уже ничего не закрываем), и только потом на открыть.

Это баг и не лучший UX, поэтому предлагаю исправить.

По поводу 100 ms. Да, в прод так делать не станешь, но в рамках демки, чтобы быстро исправить этот недочет с кнопками, на мой взгляд пойдет. Если у тебя есть идея как можно "отлепить" кнопку без задержки (а 100ms тут для этого), и желание это кодить, то я только приветствую и всячески за.